### ISSUE TITLE
remove Mapping and SymmetricMapping

### ISSUE THREAD

**Issue by arekbulski** [construct/construct#642]
@yannayl
Could you confirm if anyone uses those?

**Comment by yannayl** [comment]
Will do tomorrow

**Comment by yannayl** [comment]
pyelftools, which is quite popular has a copy of the construct 2.5. So whenever doing a search, I get tons of projects that embedded this project but not really using these feature. (I searched for SymmetricMapping, searching for Mapping gives waaay to many results.)
Anyway, I did find a few projects who actually uses it:
https://github.com/Flameeyes/glucometerutils/blob/67a70d1a544c76b5352f1dcf297c93599ab307a5/glucometerutils/support/lifescan_binary_protocol.py#L67
https://github.com/rytilahti/python-yeelightbt/blob/60dffced37c37b4f9565bfec6a04e739ac2b6370/yeelightbt/structures.py#L135
Tl;Dr - they map 'True' and 'False' to some values to some non-trivial values or something that can probably be implemented better with other features.
I suggest you keep it though. I don't think it's dead-code.

**Comment by arekbulski** [comment]
**pyelftools:**
RepeatUntilExcluding and VarIntLE/BE should be imported.
Interesting [preserve_stream_pos](https://github.com/eliben/pyelftools/blob/master/elftools/common/utils.py#L87)

**glucometerutils:**
They pinned to Constrct 2.8 because I quote "API is instable" :grin: 
They even have `construct_extras.Timestamp` ha ha ha!
I can tell they are using things like Default and  RawCopy so its not 2.5-stale, and Mapping is just IfThenElse/Flag. 

**python-yeelightbt:**
They are also using custom timestamp adapter.
And also using Embedded(Switch).

**Comment by arekbulski** [comment]
It seems that glucometerutils actually found a valid usecase for SM.
I decided to remove Mapping and rename SymmetricMapping to Mapping. All classes are excepted to be symmetrical (re-parse and re-build).

**Comment by yannayl** [comment]
Sounds reasonable.

**Comment by arekbulski** [comment]
Attribute members should be removed from Mapping, while at it.

**Comment by arekbulski** [comment]
SymmetricMapping should also be compilable.

**Comment by arekbulski** [comment]
Finished.
